5.9.0

AstreaTSS opened the pull request:
## Pull Request Type
<!-- Check the appropriate option -->

- [x] Feature addition
- [x] Bugfix
- [x] Documentation update
- [ ] Code refactor
- [ ] Tests improvement
- [ ] CI/CD pipeline enhancement
- [ ] Other: [Replace with a description]

## Description
It's a versioning merge.

Note: because I can't edit the stable branch directly, I won't be able to do what Polls and edit the version number after the merge - instead, it's likely best to wait for reviews, then quickly add the version bump commit to unstable before merging. #1513 can do this.

## Changes
<!-- List the changes you have made in a bullet-point format -->


## Related Issues
<!-- If this PR is related to any open issues, please mention them using "#ISSUENUMBER" -->


## Test Scenarios
<!-- Provide clear instructions on how to test your changes, where applicable - if no tests are required, explain why -->


## Python Compatibility
<!-- Testing 3.11 is not strictly required, but it would be nice if you could confirm that it works. -->
- [ ] I've ensured my code works on Python `3.10.x`
- [ ] I've ensured my code works on Python `3.11.x`


## Checklist
<!-- If you have not completed all of the following, there is a good chance your PR will not be merged. -->
- [ ] I've run the `pre-commit` code linter over all edited files
- [ ] I've tested my changes on supported Python versions
- [ ] I've added tests for my code, if applicable
- [ ] I've updated / added  documentation, where applicable

kennhh committed:
feat: add args, kwargs to task (#1478)

* modified:   interactions/models/internal/tasks/task.py

* reset to set_last_call_time

Signed-off-by: kennhh <133614589+kennhh@users.noreply.github.com>

---------

Signed-off-by: kennhh <133614589+kennhh@users.noreply.github.com>

i0bs committed:
feat: add id attr to app perms update event

i0bs committed:
revert: commit signoff err

i0bs committed:
ci: show pre-commit diff on failure (#1484)

AstreaTSS committed:
feat: add alt methods for multi-arg params for prefixed cmds (#1471)

* feat: add alt methods for multi-arg params for prefixed cmds

* fix: use empty if the typehint is just tuple

* ci: correct from checks.

* fix: support ConsumeRest without typehint

---------

Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>

AstreaTSS committed:
feat: cache channel data from interactions (#1479)

AstreaTSS committed:
fix: account for BaseChannel for channel mentions (#1480)

AstreaTSS committed:
feat: add default to delete/edit init interaction msg (#1481)

AstreaTSS committed:
docs: remove outdated cookiecutter template (#1483)

* docs: remove outdated cookiecutter template

* docs: add boilerplate mention

i0bs committed:
feat: add id attr to app perms update event (#1485)

LordOfPolls committed:
feat: add sort order for forums (#1488)

* feat: add forum sort order

* feat: cache layout and sort order for forums

LordOfPolls committed:
feat: add rate limit per user where needed (#1489)

Signed-off-by: LordOfPolls <dev@lordofpolls.com>

AstreaTSS committed:
fix: use message channel for PrefixedContext (#1491)

AstreaTSS committed:
fix: bound app_permissions for HybridContext (#1492)

AstreaTSS committed:
feat: add start_time(s) for AutoShardedClient (#1482)

* feat: add start_time(s) for AutoShardedClient

* feat: make start_times a dict

* docs: make start_times docstring clearer

Co-authored-by: Sophia <41456914+i0bs@users.noreply.github.com>
Signed-off-by: Astrea <25420078+AstreaTSS@users.noreply.github.com>

---------

Signed-off-by: Astrea <25420078+AstreaTSS@users.noreply.github.com>
Co-authored-by: Sophia <41456914+i0bs@users.noreply.github.com>

AstreaTSS committed:
fix: correct listen typehint (#1495)

i0bs committed:
chore: move mutable class ref/attrs to ClassVar typing (#1497)

pre-commit-ci[bot] committed:
ci: weekly check. (#1460)

* ci: weekly check.

updates:
- https://github.com/charliermarsh/ruff-pre-commit → https://github.com/astral-sh/ruff-pre-commit
- [github.com/astral-sh/ruff-pre-commit: v0.0.272 → v0.0.278](https://github.com/astral-sh/ruff-pre-commit/compare/v0.0.272...v0.0.278)
- [github.com/psf/black: 23.3.0 → 23.7.0](https://github.com/psf/black/compare/23.3.0...23.7.0)

* ci: correct from checks.

---------

Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>

AstreaTSS committed:
feat: add ability to use different arg name than option name (#1493)

* feat: add ability to use different arg name than option name

* feat: raise better error if arg name doesnt exist

* fix: use ValueError instead of TypeError

TypeError seems to be suppressed

AstreaTSS committed:
fix: make sure exported auto_defer is deco, not module (#1496)

AstreaTSS committed:
fix/feat: adjust invite obj to respect event variants (#1500)

* fix/feat: adjust invite obj to respect event variants

* docs: update docs for event object

* revert: keep MISSING for backwards compatibility

* style: consistency

Co-authored-by: Sophia <41456914+i0bs@users.noreply.github.com>
Signed-off-by: Astrea <25420078+AstreaTSS@users.noreply.github.com>

---------

Signed-off-by: Astrea <25420078+AstreaTSS@users.noreply.github.com>
Co-authored-by: Sophia <41456914+i0bs@users.noreply.github.com>

AstreaTSS committed:
fix: address issues with tag usage for guild forums (#1499)

* fix: address issues with tags for guild forums

* fix: is this breaking? better be safe than sorry

AstreaTSS committed:
docs💥: revamp extension guide (#1494)

* docs💥: revamp extension guide

* docs: update extension example

* docs: oops, forgot a word

Co-authored-by: Max <maxyolo01.ytb@gmail.com>
Signed-off-by: Astrea <25420078+AstreaTSS@users.noreply.github.com>

* docs: add section for loading all extensions in folder

* docs: turns out ipy can load multi exts in one file

* docs: lots of wording adjustments

---------

Signed-off-by: Astrea <25420078+AstreaTSS@users.noreply.github.com>
Co-authored-by: Max <maxyolo01.ytb@gmail.com>
Co-authored-by: LordOfPolls <dev@lordofpolls.com>

Damego committed:
feat: add `guild` & `channel` properties to `ThreadMembersUpdate` (#1504)

pre-commit-ci[bot] committed:
ci: weekly check. (#1503)

* ci: weekly check.

updates:
- [github.com/astral-sh/ruff-pre-commit: v0.0.278 → v0.0.280](https://github.com/astral-sh/ruff-pre-commit/compare/v0.0.278...v0.0.280)

* ci: correct from checks.

---------

Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>

Donbur4156 committed:
fix(http): incorrect path for delete permission endpoint (#1506)

* fix delete_permissions api endpoint

fix API Endpoint in delete_channel_permission
adding "permissions/" in Endpoint

* ci: correct from checks.

---------

Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>

pre-commit-ci[bot] committed:
ci: weekly check. (#1511)

updates:
- [github.com/astral-sh/ruff-pre-commit: v0.0.280 → v0.0.281](https://github.com/astral-sh/ruff-pre-commit/compare/v0.0.280...v0.0.281)

Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>

Damego committed:
feat: Implement missing stuff for scheduled events (#1507)

* feat: Implement gateway support for scheduled events

* fix: add optional to user add/remove

* refactor: pre-commit'ed

* refactor: change attrs to props and add `attrs.field`

* refactor: set repr to True

* feat(client): add `get_scheduled_event` helper method

* refactor: use cache helpers in methods

* chore: replace typing

i0bs commented:
Waiting for more reviews, I can take care of the version bump pending merge

i0bs committed:
docs: clarify intents in example (#1516)

i0bs committed:
feat: infer modal/component callback names from coroutine (#1519)

* feat: infer callback decor from coroutine name

* docs: add new logic notation

* chore: add doc notation to component callback definition

i0bs commented on interactions/api/events/discord.py:
```diff
@@ -750,3 +762,56 @@ class GuildAuditLogEntryCreate(GuildEvent):
+class GuildScheduledEventUpdate(BaseEvent):
+    """Dispatched when scheduled event is updated"""
+
+    before: Absent["ScheduledEvent"] = attrs.field(repr=True)
```
Just noticed this, should we be placing a `MISSING` default on the field?

AstreaTSS replied:
We should indeed if we're marking it as `Absent`, though I'm more curious about why it's marked that in the first place. A PR attempting to fix it should look at that.

i0bs replied:
Gave a quick look, we have absence on `before` because it's collected via. cache retrieval - we might have cache misses, ~~I'll add the default in~~ defaulting is being done through our processors, this is good to go.

i0bs commented on docs/src/Guides/20 Extensions.md:
```diff
@@ -1,248 +1,581 @@
-        from interactions.models.listener import listen
-        from interactions.ext import prefixed_commands
-        from interactions.ext.prefixed_commands import prefixed_command
+        bot = Client()
```
`Client()` instantiation defaults intents to `DEFAULT`, which doesn't include the `MESSAGE_CONTENT` privileged intent. This should be updated

i0bs replied:
(We have a message creation example in that code snippet just to clarify)

i0bs commented on docs/src/Guides/20 Extensions.md:
```diff
@@ -1,248 +1,581 @@
-        from interactions.models.events import Component
-        from interactions.models.listener import listen
-        from interactions.ext import prefixed_commands
+        bot = Client()
```
Same as per above

i0bs commented on docs/src/Guides/20 Extensions.md:
```diff
@@ -1,248 +1,581 @@
+        def __init__(self, bot):
+            self.bot = bot
+
+    def setup(bot):
```
I believe we made the setup declaration optional(?)
```suggestion
    # This is optional, but you can specify setup steps for extension loading if you wish
    def setup(bot):
```

AstreaTSS replied:
That's on purpose - the context surrounding it even notes that it's unnecessary.

i0bs commented on docs/src/Guides/20 Extensions.md:
```diff
@@ -1,248 +1,581 @@
+
+    class MyExtension(Extension):
+        @event(Startup)
+        async def startup(self):
```
If the startup declaration is optional for non-async, should we clarify how it's used here?

AstreaTSS replied:
I mean, it's noted as being one of many options you can use in the first place, I think that's good enough.

i0bs commented on docs/src/Guides/20 Extensions.md:
```diff
@@ -1,248 +1,581 @@
+import glob
+
+# replace "exts" with your folder name
+ext_filenames = glob.glob("exts/**/*.py")
```
lol `glob.glob` 🐟

AstreaTSS replied:
`datetime.datetime`

i0bs committed:
docs: clarify intents for extensions

pre-commit-ci[bot] committed:
ci: weekly check. (#1521)

updates:
- [github.com/astral-sh/ruff-pre-commit: v0.0.281 → v0.0.282](https://github.com/astral-sh/ruff-pre-commit/compare/v0.0.281...v0.0.282)

Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>

i0bs commented:
RFM upon check pass
